lib/client.js

```diff
@@ -1450,8 +1453,10 @@ function write (client, request) {
         }
 
         bytesWritten += len
-
-        if (!socket.write(chunk) && this.pause) {
+        const endCallback = request.onBodyChunkSent
+          ? request.onBodyChunkSent.bind(request, len, bytesWritten)
+          : undefined
```

Comment:
Better to just capture bytesWritten, that way you don't need to create a closure for every chunk.

Reply:
Originally, that's what I did but It didn't pass tests, as the callback is called after the socket actually writes, but `bytesWritten` is updated when `socket.write` is called. As I said in the OP, it might be easier to call `onBodyChunkSent` synchronously with the call to write.